ramips: add support for TP-Link RE205 v3

slo617 opened the pull request:
TP-Link RE205 v3 is a wireless range extender with Ethernet and 2.4G and 5G WiFi with external antennas.
It's based on MediaTek MT7628AN+MT7610EN like the RE200 v3/v4 but with external antennas.

Specifications
--------------

- MediaTek MT7628AN (580 Mhz)
- 64 MB of RAM
- 8 MB of FLASH
- 2T2R 2.4 GHz and 1T1R 5 GHz
- 1x 10/100 Mbps Ethernet
- 5x LED (GPIO-controlled), 2x button
- UART connection pads on PCB (57600 8n1)

There are 2.4G and 5G LEDs in blue which are controlled separately.

Installation
------------

Installation is identical to RE200 v3 devices as described at https://openwrt.org/toh/tp-link/re200#installation

Web Interface
-------------

It is possible to upgrade to OpenWrt via the web interface. Simply flash the -factory.bin from OEM. In contrast to a stock firmware, this will not overwrite U-Boot.

Recovery
--------

U-Boot seems to be locked on newer versions, if not it can be accessed over the UART as described in the link above.

slo617 committed:
ramips: add support for TP-Link RE205 v3

TP-Link RE205 v3 is a wireless range extender with Ethernet and 2.4G and 5G
WiFi with external antennas.
It's based on MediaTek MT7628AN+MT7610EN like the RE200 v3/v4 but with
external antennas.

Specifications
--------------

- MediaTek MT7628AN (580 Mhz)
- 64 MB of RAM
- 8 MB of FLASH
- 2T2R 2.4 GHz and 1T1R 5 GHz
- 1x 10/100 Mbps Ethernet
- 5x LED (GPIO-controlled), 2x button
- UART connection holes on PCB (57600 8n1)

There are 2.4G and 5G LEDs in blue which are controlled separately.

Installation
------------

Installation is identical to RE200 v3 devices as described at
https://openwrt.org/toh/tp-link/re200#installation

Web Interface
-------------

It is possible to upgrade to OpenWrt via the web interface. Simply flash
the -factory.bin from OEM. In contrast to a stock firmware, this will not
overwrite U-Boot.

Recovery
--------

U-Boot seems to be locked on newer versions, if not it can be accessed over
the UART as described in the link above.

Signed-off-by: Steffen Loley <slo-src@web.de>

rmilecki commented on target/linux/ramips/dts/mt7628an_tplink_re205.dtsi:
```diff
@@ -0,0 +1,166 @@
+		};
+
+		wifi5g {
+			label = "blue:wifi5g";
```
Please use `color` and `function` (`LED_FUNCTION_WLAN_5GHZ`)

rmilecki commented on target/linux/ramips/dts/mt7628an_tplink_re205.dtsi:
```diff
@@ -0,0 +1,166 @@
+		};
+
+		wifi2g {
+			label = "blue:wifi2g";
```
Please use `color` and `function` (`LED_FUNCTION_WLAN_2GHZ`)

slo617 commented:
@rmilecki 
Thank you for the review.
I changed the LED definition as requested.
I also rebased to the current main branch.

svanheule commented:
`tools/firmware-utils` has been updated to the latest version with commit 2ac6fe6a8849e700012d17705c6252716a0e8c8b, so you can drop that update from this patch. (Which as an independent changes should've been be in separate patch)

hauke commented on target/linux/ramips/dts/mt7628an_tplink_re205-v3.dts:
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7628an_tplink_re205.dtsi"
```
Why do you add a separate dtsi file?
If there is only one device using the `mt7628an_tplink_re205.dtsi` file, I would prefer if you incldue the code in the dts file. 

Do you plan to add support for more similar devices?

slo617 replied:
I just tried to copy the style of the RE200 integration.

I removed the `mt7628an_tplink_re205.dtsi` file. As the hardware is almost identical I use the `mt7628an_tplink_re200.dtsi` instead now.

slo617 commented:
As `tools/firmware-utils` have been updated with commit https://github.com/openwrt/openwrt/commit/2ac6fe6a8849e700012d17705c6252716a0e8c8b, I reverted my changes and rebased to that commit instead.
